stdlib/influxdata/influxdb/monitor/state_changes_test.flux

```diff
@@ -44,13 +28,13 @@ outData = "
 ,,2,000000000000000a,cpu threshold check,warn,statuses,whoa!,cpu,1527018860000000000,2018-05-22T19:54:22Z,threshold,vaaa,vbbb,cpu-total,host.local,7.05
 "
 
-t_state_changes = (table=<-) => table
+t_state_changes_any_to_warn = (table=<-) => table
     |> range(start: -1m)
-    |> v1.fieldsAsCols()
```

Comment:
Will these tests work on influxdb? Flux just uses csv, but influxdb will actually write the input data to storage using `to`, and I'm not sure `to` can write data that doesn't have a `_field` and `_value` column.

Reply:
Good question. I know that we do `v1.fieldsAsCols()` in the flux that we generate for thresholds.

https://github.com/influxdata/influxdb/blob/677e836bf4e2d1d3a056c2c7de31bce6b6065df3/notification/check/threshold_test.go#L108-L116
```
data
	|> v1.fieldsAsCols()
	|> monitor.check(
		data: check,
		messageFn: messageFn,
		ok: ok,
		info: info,
		warn: warn,
		crit: crit,
  )
```

and [`write` in `monitor` does](https://github.com/influxdata/flux/blob/6c04c73a47c2d0b6f2c84e584f441aaead82190a/stdlib/influxdata/influxdb/monitor/monitor.flux#L10)
```
option write = (tables=<-) => tables |> experimental.to(bucket: bucket)
```
in the [monitor.check](https://github.com/influxdata/flux/blob/6c04c73a47c2d0b6f2c84e584f441aaead82190a/stdlib/influxdata/influxdb/monitor/monitor.flux#L94-L125)

Reply:
`experimental.to` expects pivoted data whereas `influxdb.to` does not. The flux end to end tests in influxdb write the input data to storage first before running the test. This is done using a call to `influxdb.to`. That's why I think the call to `v1.fieldsAsCols()` is necessary for these test cases.

Reply:
I just tested this revision on influxdb and the tests do fail:
```
--- FAIL: TestFluxEndToEnd (24.56s)
    --- FAIL: TestFluxEndToEnd/influxdata/influxdb/monitor (0.69s)
        --- FAIL: TestFluxEndToEnd/influxdata/influxdb/monitor/state_changes_any_to_any (0.06s)
            end_to_end_test.go:174: table has no _field column
        --- FAIL: TestFluxEndToEnd/influxdata/influxdb/monitor/state_changes_info_to_any (0.06s)
            end_to_end_test.go:174: table has no _field column
        --- FAIL: TestFluxEndToEnd/influxdata/influxdb/monitor/state_changes_invalid_any_to_any (0.07s)
            end_to_end_test.go:174: table has no _field column
        --- FAIL: TestFluxEndToEnd/influxdata/influxdb/monitor/state_changes (0.06s)
            end_to_end_test.go:174: table has no _field column
```
Note that error message comes from `to` expecting a `_field` column as part of `inData`. I think you just have to format `inData` the way it was previously and add a call to `v1.fieldsAsCols` before `monitor.stateChanges`. Then it should be good to go.

Reply:
kk. not sure I totally follow why, but I'll do that.